Compile error with gcc 4.4 in travis ci

I found a compile error when I use gcc 4.4.7 with -std=c++0x and without boost in my travis CI environment. And I use header only library.

My travis CI configure is:
```
language: cpp

env:
  global:
    - LANG="zh_CN.UTF-8"
    - CMAKE_PREFIX=$HOME/prebuilt/cmake ;
    - CMAKE_VERSION=3.5.2 ;
    - MSGPACK_PREFIX=$HOME/prebuilt/msgpack-2.0.0 ;
    - MSGPACK_VERSION=2.0.0 ;
matrix:
  include:
    - os: linux
      dist: trusty
      sudo: required
      env: USE_CC=gcc-4.4 USE_CXX=g++-4.4
addons:
  apt:
    sources:
      - ubuntu-toolchain-r-test

    packages:
      - g++-4.4
      - g++-4.4-multilib
      - gcc-4.4
      - gcc-4.4-multilib

script:
  - REPO_DIR=$PWD;
  - mkdir -p $REPO_DIR/build && cd $REPO_DIR/build ;
  - cmake .. -DCMAKE_C_COMPILER=$USE_CC -DCMAKE_CXX_COMPILER=$USE_CXX
```

I have removed useless configure items. And log is here

```
In file included from /home/travis/prebuilt/msgpack-2.0.0/include/msgpack/unpack.hpp:17,

                 from /home/travis/prebuilt/msgpack-2.0.0/include/msgpack.hpp:15,

                 from /home/travis/build/atframework/libatbus/include/detail/libatbus_protocol.h:11,

                 from /home/travis/build/atframework/libatbus/include/atbus_connection.h:30,

                 from /home/travis/build/atframework/libatbus/include/atbus_endpoint.h:28,

                 from /home/travis/build/atframework/libatbus/include/atbus_node.h:34,

                 from /home/travis/build/atframework/libatbus/src/atbus_msg_handler.cpp:9:

/home/travis/prebuilt/msgpack-2.0.0/include/msgpack/v2/unpack.hpp:1265: error: declaration of ‘typedef class msgpack::v2::parser<msgpack::v2::unpacker, msgpack::v2::zone_push_finalizer> msgpack::v2::unpacker::parser’

/home/travis/prebuilt/msgpack-2.0.0/include/msgpack/v2/unpack.hpp:873: error: changes meaning of ‘parser’ from ‘class msgpack::v2::parser<msgpack::v2::unpacker, msgpack::v2::zone_push_finalizer>’

make[2]: *** [src/CMakeFiles/atbus.dir/atbus_msg_handler.cpp.o] Error 1

make[2]: *** Waiting for unfinished jobs....

In file included from /home/travis/prebuilt/msgpack-2.0.0/include/msgpack/unpack.hpp:17,

                 from /home/travis/prebuilt/msgpack-2.0.0/include/msgpack.hpp:15,

                 from /home/travis/build/atframework/libatbus/include/detail/libatbus_protocol.h:11,

                 from /home/travis/build/atframework/libatbus/include/atbus_connection.h:30,

                 from /home/travis/build/atframework/libatbus/src/atbus_connection.cpp:14:

/home/travis/prebuilt/msgpack-2.0.0/include/msgpack/v2/unpack.hpp:1265: error: declaration of ‘typedef class msgpack::v2::parser<msgpack::v2::unpacker, msgpack::v2::zone_push_finalizer> msgpack::v2::unpacker::parser’

/home/travis/prebuilt/msgpack-2.0.0/include/msgpack/v2/unpack.hpp:873: error: changes meaning of ‘parser’ from ‘class msgpack::v2::parser<msgpack::v2::unpacker, msgpack::v2::zone_push_finalizer>’

make[2]: *** [src/CMakeFiles/atbus.dir/atbus_connection.cpp.o] Error 1

In file included from /home/travis/prebuilt/msgpack-2.0.0/include/msgpack/unpack.hpp:17,

                 from /home/travis/prebuilt/msgpack-2.0.0/include/msgpack.hpp:15,

                 from /home/travis/build/atframework/libatbus/include/detail/libatbus_protocol.h:11,

                 from /home/travis/build/atframework/libatbus/include/atbus_connection.h:30,

                 from /home/travis/build/atframework/libatbus/include/atbus_endpoint.h:28,

                 from /home/travis/build/atframework/libatbus/src/atbus_endpoint.cpp:13:

/home/travis/prebuilt/msgpack-2.0.0/include/msgpack/v2/unpack.hpp:1265: error: declaration of ‘typedef class msgpack::v2::parser<msgpack::v2::unpacker, msgpack::v2::zone_push_finalizer> msgpack::v2::unpacker::parser’

/home/travis/prebuilt/msgpack-2.0.0/include/msgpack/v2/unpack.hpp:873: error: changes meaning of ‘parser’ from ‘class msgpack::v2::parser<msgpack::v2::unpacker, msgpack::v2::zone_push_finalizer>’

make[2]: *** [src/CMakeFiles/atbus.dir/atbus_endpoint.cpp.o] Error 1

In file included from /home/travis/prebuilt/msgpack-2.0.0/include/msgpack/unpack.hpp:17,

                 from /home/travis/prebuilt/msgpack-2.0.0/include/msgpack.hpp:15,

                 from /home/travis/build/atframework/libatbus/include/detail/libatbus_protocol.h:11,

                 from /home/travis/build/atframework/libatbus/include/atbus_connection.h:30,

                 from /home/travis/build/atframework/libatbus/include/atbus_endpoint.h:28,

                 from /home/travis/build/atframework/libatbus/include/atbus_node.h:34,

                 from /home/travis/build/atframework/libatbus/src/atbus_node.cpp:33:

/home/travis/prebuilt/msgpack-2.0.0/include/msgpack/v2/unpack.hpp:1265: error: declaration of ‘typedef class msgpack::v2::parser<msgpack::v2::unpacker, msgpack::v2::zone_push_finalizer> msgpack::v2::unpacker::parser’

/home/travis/prebuilt/msgpack-2.0.0/include/msgpack/v2/unpack.hpp:873: error: changes meaning of ‘parser’ from ‘class msgpack::v2::parser<msgpack::v2::unpacker, msgpack::v2::zone_push_finalizer>’

make[2]: *** [src/CMakeFiles/atbus.dir/atbus_node.cpp.o] Error 1

make[1]: *** [src/CMakeFiles/atbus.dir/all] Error 2

make: *** [all] Error 2
```

Repository is here: https://github.com/atframework/libatbus
Detail log is here: https://travis-ci.org/atframework/libatbus/jobs/142648189

Thanks

--- a/include/msgpack/v2/unpack.hpp
+++ b/include/msgpack/v2/unpack.hpp
@@ -1262,13 +1262,13 @@ struct zone_push_finalizer {
 
 class unpacker : public parser<unpacker, zone_push_finalizer>,
                  public detail::create_object_visitor {
-    typedef parser<unpacker, zone_push_finalizer> parser;
+    typedef parser<unpacker, zone_push_finalizer> parser_t;
 public:
     unpacker(unpack_reference_func f = &unpacker::default_reference_func,
              void* user_data = MSGPACK_NULLPTR,
              std::size_t initial_buffer_size = MSGPACK_UNPACKER_INIT_BUFFER_SIZE,
              unpack_limit const& limit = unpack_limit())
-        :parser(m_finalizer, initial_buffer_size),
+        :parser_t(m_finalizer, initial_buffer_size),
          detail::create_object_visitor(f, user_data, limit),
          m_z(new msgpack::zone),
          m_finalizer(*m_z) {
@@ -1318,7 +1318,7 @@ private:
 };
 
 inline bool unpacker::next(msgpack::object_handle& result, bool& referenced) {
-    bool ret = parser::next();
+    bool ret = parser_t::next();
     if (ret) {
         referenced = detail::create_object_visitor::referenced();
         result.zone().reset( release_zone() );

